src/examples/ExamplesPanel.ts

```diff
@@ -82,41 +82,48 @@ export class ExamplesPlanel {
 
     this.panel.onDidDispose(() => this.dispose(), null, this.disposables);
 
-    this.panel.webview.onDidReceiveMessage((message) => {
+    this.panel.webview.onDidReceiveMessage(async (message) => {
       switch (message.command) {
         case "openExampleProject":
           if (message.project_path && message.name) {
-            vscode.window
-              .showOpenDialog({
-                canSelectFolders: true,
-                canSelectFiles: false,
-                canSelectMany: false,
-              })
-              .then((selectedFolder: vscode.Uri[]) => {
-                if (selectedFolder && selectedFolder[0].fsPath) {
-                  const resultFolder = path.join(
-                    selectedFolder[0].fsPath,
-                    message.name
-                  );
-                  ensureDir(resultFolder)
-                    .then(() => {
-                      utils.copyFromSrcProject(
-                        message.project_path,
-                        resultFolder
-                      );
-                      const projectPath = vscode.Uri.file(resultFolder);
-                      vscode.commands.executeCommand(
-                        "vscode.openFolder",
-                        projectPath
-                      );
-                    })
-                    .catch((err) => {
-                      Logger.errorNotify("Error copying ESP-IDF example", err);
-                    });
-                } else {
-                  vscode.window.showInformationMessage("No folder selected");
-                }
-              });
+            const selectedFolder = await vscode.window.showOpenDialog({
+              canSelectFolders: true,
+              canSelectFiles: false,
+              canSelectMany: false,
+            });
+            if (!selectedFolder) {
+              vscode.window.showInformationMessage("No folder selected");
```

Comment:
Also IMO, this is redundant, since if the user don't select anything we should not perform anything and return silently, no need to be verbose.